presto-main/src/test/java/com/facebook/presto/sql/query/TestFilteredAggregations.java

```diff
@@ -86,4 +115,68 @@ public void testGroupingSets()
                         "(2, BIGINT '4', BIGINT '1'), " +
                         "(CAST(NULL AS INTEGER), BIGINT '5', BIGINT '2')");
     }
+
+    @Test
+    public void rewriteAddFilterWithMultipleFilters()
+    {
+        assertPlan(
+                "SELECT sum(totalprice) FILTER(WHERE totalprice > 0), sum(custkey) FILTER(WHERE custkey > 0) FROM orders",
+                anyTree(
+                        aggregation(
+                                ImmutableMap.of("sum_3", functionCall("sum", ImmutableList.of("sum_15")),
+                                "sum", functionCall("sum", ImmutableList.of("sum_14"))),
+                                exchange(
+                                        aggregation(
+                                                ImmutableMap.of("sum_14", functionCall("sum", ImmutableList.of("totalprice")),
+                                        "sum_15", functionCall("sum", ImmutableList.of("custkey"))),
+                                        project(
+                                                filter(
+                                                        "(\"totalprice\" > 0E0 OR \"custkey\" > BIGINT '0')",
+                                                        tableScan(
+                                                                "orders", ImmutableMap.of("totalprice", "totalprice",
+                                                                "custkey", "custkey")))))))));
+    }
+
+    @Test
+    public void testDoNotPushdownPredicateIfNonFilteredAggregateIsPresent()
+    {
+        assertPlan(
+                "SELECT sum(totalprice) FILTER(WHERE totalprice > 0), sum(custkey) FROM orders",
+                anyTree(
```

Comment:
instead you can use something like: com.facebook.presto.sql.planner.TestLogicalPlanner#assertPlanContainsNoApplyOrAnyJoin to see if there is no filter node